takari-lifecycle-plugin/src/test/java/io/takari/maven/plugins/compile/jdt/CompileJdtClasspathTest.java

```diff
@@ -142,6 +145,29 @@ public void testRepository_classpathOrder() throws Exception {
     mojos.assertBuildOutputs(parent, "module-a/target/classes/reactor/modulea/ModuleA.class");
   }
 
+  @Test
+  public void testReactorUnused() throws Exception {
+    File parent = resources.getBasedir("compile-jdt-classpath/reactor-unused");
+    Xpp3Dom unused = new Xpp3Dom("unusedDeclaredDependency");
+    unused.setValue("error");
+
+    mojos.flushClasspathCaches();
+
+    File moduleB = new File(parent, "module-b");
+
+    mojos.compile(moduleB, unused);
+
+    MavenProject moduleA = mojos.readMavenProject(new File(parent, "module-a"));
+    addDependency(moduleA, "module-b", new File(moduleB, "target/classes"));
+
+    try {
+      mojos.compile(moduleA, unused);
+      fail();
+    } catch (MojoExecutionException e) {
+      assertTrue(e.getLocalizedMessage().contains("The following dependencies are declared but are not used: [test:module-b:jar:1.0:compile]"));
```

Comment:
Don't assert the exception, use `CompileRule.assertMessages` to assert expected message was created on the pom file.